### PR TITLE
Jenkins caf module

### PR DESCRIPTION
**Pull request by kavithai**

### CODE REVIEW

File: modules/jenkins/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# GitLab CAF Module
```
Comment (hattan) [inline modules/jenkins/README.md]: Shouldn't this be the Jenkins CAF Module?

File: modules/jenkins/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+This submodule is part of [Cloud Adoption Framework](https://github.com/aztfmod/terraform-azurerm-caf) landing zones for [GitLab on Terraform](https://github.com/taiidani/terraform-provider-jenkins).
```
Comment (hattan) [inline modules/jenkins/README.md]: There is another reference to gitlab in the link text

File: modules/jenkins/project.tf [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Create a Jenkins Folder if foldername!="" and createFolder="true"
+resource "jenkins_folder" "job" {
+  name               = lookup(var.job, "folderName", "")
+  description        = lookup(var.job, "folderDescription", "")
+  count              = (var.job.folderName!="" && var.job.createFolder=="true")? "1":"0"
```
Comment (hattan) [inline modules/jenkins/project.tf:5]: 👍 very cool!

**Review by hattan — CHANGES_REQUESTED** [review]
Looks great, just two small doc changes to replace references to gitlab with jenkins

**Comment by hattan** [comment]
Due to the current architecture, we need to include the JENKINS variables in the other ci workflows like this
https://github.com/aztfmod/terraform-addons-caf/blob/jenkinsCafModule/.github/workflows/ci_gitlab.yml#L105

When the refactor to break them apart takes place we can then remove the need for these.

Can you please add the missing jenkins env variables to:

- [ ] gitlab ci
- [ ] github ci
- [ ] azdo ci

File: .github/workflows/ci_jenkins.yml [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+          export ARM_SUBSCRIPTION_ID=$(echo "${AZURE_CREDENTIALS}" | jq -r ".subscriptionId")
+          export ARM_TENANT_ID=$(echo "${AZURE_CREDENTIALS}" | jq -r ".tenantId")
+
+          ENVIRONMENT_SUFFIX=${{ github.run_id }}
+
+          export environment="ci_${ENVIRONMENT_SUFFIX}"
+
+          echo "Environment name is 'ci_${ENVIRONMENT_SUFFIX}'"
+
+          # deploy jenkins
```
Comment (in4margaret) [inline .github/workflows/ci_jenkins.yml:96]: Please specify what exactly get deployed

File: modules/jenkins/variables.tf [hunk 5]
```diff
@@ -0,0 +1,3 @@
+variable job {
+  description = "The object representative of the gitlab project entity to update"
```
Comment (in4margaret) [inline modules/jenkins/variables.tf]: gitlab -> jenkins?

File: .github/workflows/ci_jenkins.yml [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+          export ARM_SUBSCRIPTION_ID=$(echo "${AZURE_CREDENTIALS}" | jq -r ".subscriptionId")
+          export ARM_TENANT_ID=$(echo "${AZURE_CREDENTIALS}" | jq -r ".tenantId")
+
+          ENVIRONMENT_SUFFIX=${{ github.run_id }}
+
+          export environment="ci_${ENVIRONMENT_SUFFIX}"
+
+          echo "Environment name is 'ci_${ENVIRONMENT_SUFFIX}'"
+
+          # destroy launchpad
```
Comment (in4margaret) [inline .github/workflows/ci_jenkins.yml:149]: Can you run destroy even when previous steps errored out?

**Review by in4margaret — CHANGES_REQUESTED** [review]

**Comment by DariuszPorowski** [comment]
I am going to close this PR, because this module has been adjusted to new structure and published under this PR: https://github.com/aztfmod/terraform-addons-caf/pull/13

**Pull request closed without merging**